ratis-server/src/main/java/org/apache/ratis/server/DataStreamServerRpc.java

```diff
@@ -27,6 +27,11 @@
    */
   void startServer();
 
+  /**
+   * start clients that used to forward requests to peers.
+   */
+  void startClientToPeers();
```

Comment:
Have this API because we need to have all stream servers setup, then for those servers that need to forward message, call `startClientToPeers` to initialize clients connecting to peers.

Reply:
startServer should take care the initial setup.  Would there be a case that startServer will be called but not startClientToPeers?

Reply:
`startServer` is to start the stream server.  `startClientToPeers` means to build connections with other stream servers, which implies that other stream servers must be called `startServer` firstly.

That's why I cannot call `startClientToPeers` in `startServer` because other stream servers might not start.

Alternatively, we could lazy initialize the clients to peers, by checking whether connection is built when having first request. But I am not sure whether this is a good idea.

 `startServer` can be called without calling `startClientToPeers` afterwards, which means that this stream will not forward message to peers.

Reply:
> startServer can be called without calling startClientToPeers afterwards, which means that this stream will not forward message to peers.

I think `startClientToPeers` should put into `startServer`, even though this stream will not forward message to peers,  because in this case clients is empty.

Reply:
The problem is when `startServer`, as other servers might not ready, we cannot guarantee that we can connect to peer. 

If you still want to hide this `startClientToPeers`, then maybe we should do some lazy connection that when receives a first request, connects with other servers.

Reply:
lazy connection is OK.

Reply:
The correct way to make startClient to retry.  In a distributed system, we cannot assume that all the servers are already started.

It needs more works for retrying.  I am fine to add NettyServerStreamRpc.startClientToPeers as a work around in the mean time.  However, please remove DataStreamServerRpc.startClientToPeers.

Reply:
Ok I have chosen to use the workaround that exposes `NettyServerStreamRpc.startClientToPeers` , and file https://issues.apache.org/jira/browse/RATIS-1099 to track make the connection built automatically without a call.